crates/shared/src/http_solver_api/model.rs

```diff
@@ -130,6 +130,15 @@ pub struct FeeModel {
     pub token: H160,
 }
 
+#[serde_as]
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub struct InteractionData {
+    pub target: H160,
+    pub value: U256,
+    pub call_data: Bytes<Vec<u8>>,
+    pub exec_plan: Option<ExecutionPlanCoordinatesModel>,
```

Comment:
I was surprised this is optional (should be at least commented). 
The reason it is optional in the other model is because we may not always be able to derive an execution plan for once we found the hollistic solution. Does this make sense here, too? Thinking about it more long term I think we will need to force solvers to come up with an execution plan (as otherwise it's hard to convert a solution into calldata deterministically)

Reply:
Would still like to get an answer to this comment.

Reply:
I agree that long-term we want to have this non-optional. But in the current code, I prefer to keep it the same as for the usual http solver. Otherwise, the asymmetric handling of option and non-option makes the code look quite bad.
How about: I make a follow-up PR that enforces it for both alternatives, the execution plan is non-optional. 

>The reason it is optional in the other model is because we may not always be able to derive an execution plan for once we found the hollistic solution

I even don't really fully understand the optionality in this case: If the solver can not come up with an execution plan, will the driver be able to do it? This work should definitively happen on the solvers side.

Reply:
Maybe @marcovc can weigh in here how realistic it is that the solver doesn't have an execution plan for AMMs.

No need to change if you prefer the symmetry, however I think that thanks to the `coordinates()` abstraction (which may continue to return an Option) you could just return `Some(interaction_data.exect_plan)`. I don't see how it would make the code look bad.

Reply:
Hi! We can make it non-optional (we will just return the trivial solution if for some reason we can't come up with an execution plan). The reason why is optional is that, for debugging issues in computing the execution plan, it is helpful to still see the solution to the problem.